fdbserver/MoveKeys.actor.cpp

```diff
@@ -1270,10 +1272,10 @@ ACTOR static Future<Void> startMoveShards(Database occ,
 		loop {
 			state Key begin = keys.begin;
 			state KeyRange currentKeys = keys;
-			state int maxRetries = 0;
 			state bool complete = false;
 
-			state Transaction tr(occ);
+			// RYW to optimize re-reading the same key ranges. See readTSSMappingRYW
+			state Reference<ReadYourWritesTransaction> tr = makeReference<ReadYourWritesTransaction>(occ);
```

Comment:
Do we have to use `ReadYourWritesTransaction` here? `Transaction` was used intentionally since `ReadYourWritesTransaction` is not compatible with `unassignServerKeys()`, which is called within `startMoveShards()`, currently it is not actually called since the condition is always `false`, but it could be triggered in the future.

Sorry I should have added a comment here.

Reply:
I was modeling what startMoveKeys does.  The call to readTSSMappingRYW wants a RYWTransaction. Let me go back to plain Transaction (and it looks like readTSSMappingRYW internally doesn't actually need RYWT....).